### DISCUSSION TITLE
Can spot account and margin account at one strategy?

### DISCUSSION THREAD

**Discussion by OdinLin** [gbeced/basana#10] [Q&A] [answered]
Can I both use spot account and margin account at one strategy? I've look at the example, It seen that one strategy subscribe to one trading signals, either spot or margin.

**Comment by gbeced** [comment, marked as answer]
You can use all 3 accounts at the same time if you want. The strategy, that is responsible for generating the trading signals, subscribes to the bar events and that has nothing to do with the spot/margin/isolated accounts.

**Reply by OdinLin** [reply]
Yes, but TradingSignal could not specify which account need to buy or sell, It just contain time , operation and pair.

**Reply by gbeced** [reply]
It is just a design choice for the example. That is the scope that I decided to give to that class. It has the responsibility to let you know when to buy or sell a trading pair based on some technical indicator. That is it.

When you handle that event, you may decide to use your spot account, or your leverage account, whether to use market orders, limit orders, the size of the order, etc. These are also no part of the trading signal.

But having said that, you are free to do it any other way. Note that in the [Quickstart](https://basana.readthedocs.io/en/latest/quickstart.html) y wrote **The way the examples are structured is just one way to do it. You’re free to structure the code in any other way**.
You could reuse any sample strategy and push your extended trading signal that includes whatever information you think it makes sense to be included in the event.

**Reply by OdinLin** [reply]
Thanks , I got it !
